tests/infra_test.go

```diff
@@ -303,9 +301,22 @@ var _ = Describe("Infrastructure", func() {
 			keys := getKeysFromMetrics(metrics)
 			for _, key := range keys {
 				// we don't care about the ordering of the labels
-				Expect(key).To(ContainSubstring(fmt.Sprintf("node=%s", nodeName)))
-				Expect(key).To(ContainSubstring(fmt.Sprintf("namespace=%s", vmi.Namespace)))
-				Expect(key).To(ContainSubstring(fmt.Sprintf("name=%s", vmi.Name)))
+				Expect(key).To(ContainSubstring(fmt.Sprintf("node=\"%s\"", nodeName)))
+				Expect(key).To(ContainSubstring(fmt.Sprintf("namespace=\"%s\"", vmi.Namespace)))
+				Expect(key).To(ContainSubstring(fmt.Sprintf("name=\"%s\"", vmi.Name)))
+			}
+		}, 300)
+
+		It("should include VMI telemetry for a running VM", func() {
```

Comment:
Can we check more than one VMIs so we check that the VMIs loop is working fine too ?

Reply:
It is a good idea, but I'd like to extend this to other metrics-related checks, so I'll do in another PR.

Reply:
Here: https://github.com/kubevirt/kubevirt/pull/2512

Reply:
ack, thanks!